src/Parallel/Reduction.cpp

```diff
@@ -3,9 +3,11 @@
 
 #include "Parallel/Reduction.hpp"
 
+#include "Utilities/Gsl.hpp"
+
 namespace Parallel {
-NoSection& no_section() noexcept {
+gsl::not_null<NoSection*> no_section() noexcept {
```

Comment:
Return references.  Pointers are just for arguments.

Reply:
thanks, fixed.